### ISSUE TITLE
Smart rebuild doesn't appear to respect external_links

### ISSUE THREAD

**Issue by lox** [docker/compose#1600]
Given a container like this:

```
auth:
  image: 99designs/auth
  links:
    - memcached
```

Then started: `docker-compose up -d auth`

If I then add external_links:

```
auth:
  image: 99designs/auth
  links:
    - memcached
  external_links:
    - mysql-5.6:mysql_auth
    - mysql-5.6:mysql_userlogin
```

and then run `docker-compose up -d auth`, I'd expect it to be rebuilt to add the new links. This doesn't appear to happen.

**Comment by aanand** [comment]
1. Is the `mysql-5.6` container definitely running when you run `docker-compose up -d auth` the second time?
2. Is the ID of the `auth` container the same before/after running `docker-compose up -d auth` the second time?
3. Can you paste the output of `docker inspect` on the `auth` container, as well as `docker ps`?

**Comment by dnephin** [comment]
This was a bug in 1.4.0, it was fixed in 1.4.2 (#1960)
